tests/harness/cases/optional.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package tests.harness.cases;
+option go_package = "github.com/envoyproxy/protoc-gen-validate/tests/harness/cases/go;cases";
+
+import "validate/validate.proto";
+
+message TestOptionalMsg {
+    string  const  = 1 [(validate.rules).string.const = "foo"];
+    TestOptionalMsg nested = 2;
+}
+
+message MessageRequiredButOptional { optional TestOptionalMsg val = 1 [(validate.rules).message.required = true]; }
+message Int64LTEOptional      { optional int64 val = 1 [(validate.rules).int64.lte = 64]; }
```

Comment:
Please add some test cases to the executor, otherwise this won't actually test the behavior of these: https://github.com/envoyproxy/protoc-gen-validate/blob/main/tests/harness/executor/cases.go

Reply:
i moved the tests to existing proto files since creating new ones resulted in a level of complexity i didn't want to deal with.